addons/mail/static/src/messaging/entity/entity/field.js

```diff
@@ -0,0 +1,638 @@
+odoo.define('mail.messaging.entity.Field', function (require) {
+'use strict';
+
+class Field {
```

Comment:
Eventually each entity type should have its own subclass I think, having conditions on `fieldType` does not seem very natural.

Reply:
As discussed.

Boolean, String, Number, one for each type of relation.

Array and Object can probably be avoided.